lib/hooks.js

```diff
@@ -1,40 +1,33 @@
 const RiffError = require("./riff-error");
 const { types: errorTypes } = require("./errors");
 
-const handleError = (reject) => (err) => {
-    return reject(new RiffError(errorTypes.HOOK_RUNTIME, err));
+const handleError = (err) => {
+    return new RiffError(errorTypes.HOOK_RUNTIME, err);
 };
 
-module.exports.guardWithTimeout = (hookFn, timeoutInMs) => {
-    return new Promise((resolve, reject) => {
-        const hookType = typeof hookFn;
-        if (hookType === "undefined") {
-            return resolve();
-        }
-        if (hookType !== "function") {
-            return reject(
-                new RiffError(
-                    errorTypes.HOOK_INVALID,
-                    `Hooks must be functions, found: ${hookType}`
-                )
-            );
-        }
+module.exports.guardWithTimeout = async (hookFn, timeoutInMs) => {
+    if (typeof hookFn === "undefined") {
+        return;
+    }
+    if (typeof hookFn !== "function") {
+        throw new RiffError(
+            errorTypes.HOOK_INVALID,
+            `Hooks must be functions, found: ${typeof hookFn}`
+        );
+    }
 
-        const timerId = setTimeout(() => {
-            reject(
-                new RiffError(
-                    errorTypes.HOOK_TIMEOUT,
-                    `The hook took too long to run (timeout: ${timeoutInMs}ms). Aborting now`
-                )
-            );
-        }, timeoutInMs);
-
-        try {
-            Promise.resolve(hookFn())
-                .then(resolve, handleError(reject))
-                .finally(() => clearTimeout(timerId));
-        } catch (err) {
-            handleError(reject)(err);
-        }
+    let timerId;
+    const hookPromise = Promise.resolve()
+        .then(hookFn)
+        .catch((err) => Promise.reject(handleError(err)));
+    const timerPromise = new Promise((resolve, reject) => {
+        const timeoutError = new RiffError(
+            errorTypes.HOOK_TIMEOUT,
+            `The hook took too long to run (timeout: ${timeoutInMs}ms). Aborting now`
+        );
+        timerId = setTimeout(() => reject(timeoutError), timeoutInMs);
```

Comment:
No need to create the error until it's needed
```suggestion
        timerId = setTimeout(() => {
            reject(new RiffError(
                errorTypes.HOOK_TIMEOUT,
                `The hook took too long to run (timeout: ${timeoutInMs}ms). Aborting now`
            ));
        }, timeoutInMs);
```